tests/unit/amo/components/TestInstallButtonWrapper.js

```diff
@@ -0,0 +1,239 @@
+import * as React from 'react';
+
+import GetFirefoxButton, {
+  GET_FIREFOX_BUTTON_TYPE_ADDON,
+} from 'amo/components/GetFirefoxButton';
+import InstallButtonWrapper, {
+  InstallButtonWrapperBase,
+} from 'amo/components/InstallButtonWrapper';
+import { setInstallState } from 'core/actions/installations';
+import AMInstallButton from 'core/components/AMInstallButton';
+import { CLIENT_APP_FIREFOX, INSTALLED, UNKNOWN } from 'core/constants';
+import { createInternalAddon } from 'core/reducers/addons';
+import { createInternalVersion, loadVersions } from 'core/reducers/versions';
+import {
+  createContextWithFakeRouter,
+  createFakeLocation,
+  dispatchClientMetadata,
+  fakeAddon,
+  fakeInstalledAddon,
+  shallowUntilTarget,
+  userAgentsByPlatform,
+} from 'tests/unit/helpers';
+
+describe(__filename, () => {
+  let store;
+
+  beforeEach(() => {
+    store = dispatchClientMetadata().store;
+  });
+
+  const render = (props = {}) => {
+    return shallowUntilTarget(
+      <InstallButtonWrapper
+        addon={createInternalAddon(fakeAddon)}
+        defaultInstallSource="some-install-source"
+        location={createFakeLocation()}
+        store={store}
+        {...props}
+      />,
+      InstallButtonWrapperBase,
+      {
+        shallowOptions: createContextWithFakeRouter(),
+      },
+    );
+  };
+
+  const _loadVersions = ({ addon = fakeAddon } = {}) => {
+    store.dispatch(
+      loadVersions({
+        slug: addon.slug,
+        versions: [
+          {
+            ...addon.current_version,
+          },
+        ],
+      }),
+    );
+  };
+
+  const _dispatchClientMetadata = (params = {}) => {
+    return dispatchClientMetadata({
+      store,
+      userAgent: userAgentsByPlatform.mac.firefox57,
+      ...params,
+    });
+  };
+
+  it('for Firefox, calls getClientCompatibility to determine the compatibility', () => {
```

Comment:
I *thought* you were supposed to read these as a sentence (altho I am not 100% about that). So "it for Firefox, calls.." doesn't sound right. So assuming my sentence theory is right:) maybe ~ "it(' calls getClientCompatibility to determine the compatibility for firefox', () =>`

Reply:
sorry if I missed that before..

we can also get a 2nd opinion on this since like I said I am not 100% about this:)

Reply:
That's a good point. I'll fix this and the others.